### DISCUSSION TITLE
Return ent.text and ent.label_ in dictionary format

### DISCUSSION THREAD

**Discussion by musicnandpeace** [explosion/spaCy#11707] [Help: Coding & Implementations] [answered]
I created a custom NER spacy model and intend to create a single page app that takes in input and gives out json result of extracted entities. 

```python
nlp = spacy.load("my-model")
doc = nlp(''' Patient is taking dolo 100mg for fever. Complains of sleeplessness and pain in left leg. History of CKD. Underwent gall bladder laparoscopy in feb 2022.''')
```

Now to get entities and create a dictionary out of it I used the following code:

```python
entities = list(nlp.get_pipe('ner').labels)
extracted_entities = dict.fromkeys(entities, [])
```

`extracted_entities` looks like this:

```
{'DIAGNOSIS': [],
 'DRUG': [],
 'LAB TEST/ RECORDING': [],
 'PROCEDURE/TREATEMENT': [],
 'SYMPTOMS_COMPLAINTS': []}
```
 
 Now all i want to do is get the entities into their respective list in the `extracted_entities` dictionary by using this:

```python
 for ent in doc.ents:
    label = ent.label_
    extracted_entities[ent.label_].append(ent)
```
    
 But this code adds all entities in all the lists in `extracted_entities` dictionary. How can i get the resultant labels with their `ent.text` in desired dictionary / json format?

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @musicnandpeace , this may be more of how Python's [`dict.fromkeys`](https://docs.python.org/3/library/stdtypes.html?highlight=fromkeys#dict.fromkeys) works than in spaCy. One solution is to use a dictionary comprehension:

```python
extracted_entities = {ent: [] for ent in entities}
```

Also, next time please format the codeblocks using backticks (```), it helps us better debug and copy-paste your sample code

**Reply by musicnandpeace** [reply]
Replacing dict.fromkeys with dictionary comprehension is giving the proper result . Thank you so much @ljvmiranda921!
 
Sorry, I am a noob and would need some more guidance. The folder of best model I custom trained is named 'my-model'. It contains ner - toke2vec - vocab - config - meta -tokenizer files.  When i try to load it using ```nlp = spacy.load("my-model")```  in my py file, it gives the following error:
OSError: [E050] Can't find model 'my-model'. It doesn't seem to be a Python package or a valid path to a data directory.
Although it gives no such error when i run it on my jupyter notebook.  

This is what my folder structure looks like:
├── FastAPI
│   ├── __pychache__
│   ├── endo_ner_env
│   ├── ner.py
│   └── model
│   │   ├── __init__.py
│   │   ├── ner_main.py
│   │   └── my-model   (contains ner - toke2vec - vocab - config - meta -tokenizer files)
│   │   └── __pychache__


How can i resolve this problem? Thanks in advance.
